### ISSUE TITLE
signout function does not signout Google

### ISSUE THREAD

**Issue by railsstudent** [aws/amazon-cognito-auth-js#68]
I could use the SDK to authenticate by Google account without problem. After I call signout() to sign out from web app, the google account is not really signed out. 
When I attempted to login again,  the same google account is used for authentication while I wished to use a different google account.  
What is the proper way to log out of google account such that account chooser is loaded for me to select account?  
The workaround is to go to gmail.com to actually log out before returning to the app to login again.
This is not a very good user experience and introduce security issue because another person can use that computer to login without typing email and address.

**Comment by lucasgao** [comment]
clean cookies after signout

**Comment by yuntuowang** [comment]
@lucasgao is correct, browser will store the google sign in info in cookies. You need to clean cookies after sign out of your app.

**Comment by railsstudent** [comment]
Is there api to clear cookies that store google sign in token, etc?

**Comment by yuntuowang** [comment]
@railsstudent Basically cognito will clear cognito cookies in browser once you click "sign out“. But google cookies won't be cleared. You may look into google API to see if there is a way. 

We will look into this issue and work on it for sure!

**Comment by railsstudent** [comment]
I created an iframe that calls https://accounts.google.com/Logout.
A hack until a better solution is found.

**Comment by yuntuowang** [comment]
@railsstudent, awesome!

**Comment by bskinny** [comment]
Signing out of Cognito Auth should not sign the user out of Google. A user can be signed into Google beforehand and should stayed signed in after any Cognito logout. Correct?

**Comment by yuntuowang** [comment]
HI @bskinny, yes. Basically cognito will clear cognito cookies in browser once you click "sign out“. But google cookies won't be cleared.

**Comment by raphadko** [comment]
Also having this problem here.

I have been trying to log users into cognito using Google as an identity provicer for 5 days now. Every single option or library that I've tried out there is either incomplete, undocumented or broken. This library seems to work partially. Not being able to sign the user out would pretty much mean it's broken too.  (Signing the user out of google itself is not an option). 

On the issues forum for amazon-cognito-identity-js repository, people are just leaving Cognito to move on to Firebase (https://github.com/aws/amazon-cognito-identity-js/issues/508). As a newcomer to AWS services, I'm confused. Considering the importance of AWS/Cognito in the serverless ecossystem (considered by many the future of the web), and Google probably being the #1 identity provider in the world, is it really the case that using Cognito+Google just does not currently work or am I really missing something here? 

ps: if Amazon needs a product owner, I'd be happy to apply for the position. (:

**Comment by bskinny** [comment]
@raphadko Have you tried running the sample? Authenticating with the Google external identity provider works but perhaps using the sample application will determine if you problem is with AWS or Google configuration.

**Comment by raphadko** [comment]
Yes I used the sample.

**Comment by howard-lin-zocdoc** [comment]
I'm facing the same problem. What I need is to have the ability to switch Google Accounts. Even after user is logged out with Cognito, Google still remember the last account used to login. User end up with a login loop and have no option to select another account.

Google Sign In Web supports `prompt= select_account` to always show the account chooser. 

I think the feature request here is to have Cognito also support adding this query parameter when landing on Google login page.

Search for `select_account` in Google's documentation.
https://developers.google.com/identity/sign-in/web/reference#googleauthsigninoptions

**Comment by yuntuowang** [comment]
Hi @howard-lin-zocdoc, I get your point. Basically, since once you click sign-out, google cookie won't be cleared. So you won't be redirected to the google page to choose an account to sign in.

**Comment by howard-lin-zocdoc** [comment]
@yuntuowang Yup, I think there is no way we can clear cookies under Google's domain. Can you comment on if adding `prompt=select_account` this is something you guys will do?

**Comment by yuntuowang** [comment]
I will mark this as a feature request on your behalf. For sure, it will be in our feature request queue, however I cannot comment on the delivery time of it now.

**Comment by fgrillo** [comment]
@yuntuowang any news regarding this issue? I'm also experiencing this, users are not prompted to select the google account after they choose one for the first time.

**Comment by rioastamal** [comment]
@yuntuowang Any update?

**Comment by NivethaMuthukumar** [comment]
> re request on your behalf. For sure, it will be in our feature request queu

@yuntuowang any update regarding this? i'm also facing this issue

**Comment by rakesh0r** [comment]
@yuntuowang any update ? i'm also facing this issue

**Comment by chvenkataraman** [comment]
We are also facing the same issue. This is the usual requirement to go to prompt again for login after logout - instead of considering the same previous user. @yuntuowang: Can we make this as priority (by changing to some other instead of feature request)?

**Comment by jorgexgb** [comment]
Any updates on this?
